src/System.Net.Http/src/System/Net/Http/Managed/ManagedHandler.cs

```diff
@@ -0,0 +1,307 @@
+            get { return _preAuthenticate; }
+            set { CheckInUse(); _preAuthenticate = value; }
+        }
+
+        public bool UseDefaultCredentials
+        {
+            get { return _useDefaultCredentials; }
+            set { CheckInUse(); _useDefaultCredentials = value; }
+        }
+
+        public ICredentials Credentials
+        {
+            get { return _credentials; }
+            set { CheckInUse(); _credentials = value; }
+        }
+
+        public bool AllowAutoRedirect
+        {
+            get { return _allowAutoRedirect; }
+            set { CheckInUse(); _allowAutoRedirect = value; }
+        }
+
+        public int MaxAutomaticRedirections
+        {
+            get { return _maxAutomaticRedirections; }
+            set { CheckInUse(); _maxAutomaticRedirections = value; }
+        }
+
+        public int MaxConnectionsPerServer
+        {
+            get { return _maxConnectionsPerServer; }
+            set
+            {
+                CheckInUse();
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _maxConnectionsPerServer = value;
+            }
+        }
+
+        public int MaxResponseHeadersLength
+        {
+            get { return _maxResponseHeadersLength; }
+            set
+            {
+                CheckInUse();
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                _maxResponseHeadersLength = value;
+            }
+        }
+
+        public X509CertificateCollection ClientCertificates
+        {
+            get
+            {
+                if (_clientCertificateOptions != ClientCertificateOption.Manual)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.net_http_invalid_enable_first, nameof(ClientCertificateOptions), nameof(ClientCertificateOption.Manual)));
+                }
+
+                return _clientCertificates ?? (_clientCertificates = new X509Certificate2Collection());
+            }
+        }
+
+        public Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool> ServerCertificateCustomValidationCallback
+        {
+            get { return _serverCertificateCustomValidationCallback; }
+            set { CheckInUse(); _serverCertificateCustomValidationCallback = value; }
+        }
+
+        public bool CheckCertificateRevocationList
+        {
+            get { return _checkCertificateRevocationList; }
+            set { CheckInUse(); _checkCertificateRevocationList = value; }
+        }
+
+        public SslProtocols SslProtocols
+        {
+            get { return _sslProtocols; }
+            set
+            {
+                CheckInUse();
+#pragma warning disable 0618 // obsolete warning
+                if ((value & (SslProtocols.Ssl2 | SslProtocols.Ssl3)) != 0)
```

Comment:
Plus, as a framework, I don't believe we should be in the business of dictating what the customer's code can and can't do from this perspective.  They can choose to disable protocols.  And we can choose to have a default that's always as secure as we'd like.  But we shouldn't be in the business of saying "even though your business requires you to use those protocols, we're not going to let you, so go use some other dev platform if you need them".